lib/3scale_toolbox/commands/import_command/openapi/update_service_proxy_step.rb

```diff
@@ -0,0 +1,67 @@
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        class UpdateServiceProxyStep
+          include Step
+
+          ##
+          # Updates Proxy config
+          def call
+            # setting required attrs, operation is idempotent
+            proxy_settings = {}
+
+            add_api_backend_settings(proxy_settings)
+            add_security_proxy_settings(proxy_settings)
+
+            return unless proxy_settings.size.positive?
+
+            service.update_proxy proxy_settings
+            puts 'Service proxy updated'
+          end
+
+          private
+
+          def add_api_backend_settings(settings)
+            scheme = api_spec.schemes.first || 'https'
+            host = api_spec.host
+
+            settings[:api_backend] = "#{scheme}://#{host}"
+            nil
+          end
+
+          def add_security_proxy_settings(settings)
+            # nothing to add on proxy settings when no security required in openapi
+            return if security.nil?
+
+            case security.type
+            when 'oauth2'
+              settings[:credentials_location] = 'headers'
+              settings[:oidc_issuer_endpoint] = oidc_issuer_endpoint unless oidc_issuer_endpoint.nil?
+            when 'apiKey'
+              settings[:credentials_location] = credentials_location
+              settings[:auth_user_key] = security.name
+            else
+              raise ThreeScaleToolbox::Error, "Unexpected security scheme type #{security.type}"
+            end
+
+            nil
+          end
+
+          def credentials_location
+            return nil if security.nil?
```

Comment:
Why return nil over just return?

Reply:
no reason :woman_shrugging:

Reply:
actually, `security.nil?` can never happen. `credentials_location` is only used in a method that  checks previously that. I was just adding safety checking just in case that method is moved somewhere else.

Reply:
imaginary problems

Reply:
Well, my question was more about why `return nil if security.nil?` over `return unless security`, not why the check at all.